controllers/useraccount/useraccount_controller.go

```diff
@@ -302,23 +322,20 @@ func (r *Reconciler) ensureNSTemplateSet(logger logr.Logger, userAcc *toolchainv
 				return nil, false, err
 			}
 			nsTmplSet = newNSTemplateSet(userAcc)
-			if err := controllerutil.SetControllerReference(userAcc, nsTmplSet, r.Scheme); err != nil {
-				return nil, false, r.wrapErrorWithStatusUpdate(logger, userAcc, r.setStatusNSTemplateSetCreationFailed, err,
-					"failed to set controller reference for NSTemplateSet '%s'", name)
-			}
 			if err := r.Client.Create(context.TODO(), nsTmplSet); err != nil {
 				return nil, false, r.wrapErrorWithStatusUpdate(logger, userAcc, r.setStatusNSTemplateSetCreationFailed, err,
 					"failed to create NSTemplateSet '%s'", name)
 			}
-			logger.Info("NSTemplateSet created successfully", "name", name)
+			logger.Info("NSTemplateSet created successfully")
 			return nsTmplSet, true, nil
 		}
 		return nil, false, r.wrapErrorWithStatusUpdate(logger, userAcc, r.setStatusNSTemplateSetCreationFailed, err, "failed to get NSTemplateSet '%s'", name)
 	}
-	logger.Info("NSTemplateSet already exists", "name", name)
+	logger.Info("NSTemplateSet already exists")
 
 	// update if not same
-	if !reflect.DeepEqual(nsTmplSet.Spec, userAcc.Spec.NSTemplateSet) {
+	if userAcc.Spec.NSTemplateSet != nil &&
```

Comment:
if the NSTemplateSet is nil is already checked at the very beginning of this function, right?

Reply:
indeed, it's superfluous here. Removed in bb0188b